src/plugins/core/merge.ts

```diff
@@ -257,8 +257,12 @@ export class MergePlugin extends CorePlugin<MergeState> implements MergeState {
     if (!this.isInMerge(position)) {
       return position;
     }
-    const mergeTopLeftPos = this.getMerge(position)!.topLeft;
-    return { sheetId: position.sheetId, col: mergeTopLeftPos.col, row: mergeTopLeftPos.row };
+    const mergeZone = this.getMerge(position);
+    return {
+      sheetId: position.sheetId,
+      col: mergeZone?.left as HeaderIndex,
+      row: mergeZone?.top as HeaderIndex,
```

Comment:
```suggestion
      col: mergeZone!.left,
      row: mergeZone!.top,
```
No need to re-cast into HeaderIndex, you can tell typescript it has to have a value with `!`

Reply:
History has shown us that each time we introduce a `!`, a future bug is also introduce. `!` is a way to tell typescript to have truth in you. It could be correct now, but could become incorrect in the future and we will not notice it. The right way to handle this is to check that `mergeZone` is defined, and you did it correctly. So, simply remove the cast and optional:
```
col: mergeZone.left,
row: mergeZone.top,
```

Reply:
Note that `!` can be used in tests